mobile/native/src/trade/order/handler.rs

```diff
@@ -90,34 +89,28 @@ pub async fn submit_order(
 
     db::insert_order(order.clone()).map_err(SubmitOrderError::Storage)?;
 
-    let runtime = get_or_create_tokio_runtime().expect("to be ablet to get runtime");
-    if let Err(err) = runtime
-        .spawn_blocking({
-            let order = order.clone();
-            move || {
-                let mut db = db::connection()?;
-                // We need to ensure that the channel-opening parameters are inserted after the
-                // order has been posted, but before order has been match and we need them to open
-                // the DLC channel. Without this database transaction we can easily end up with a
-                // match before we insert them!
-                db.transaction(|conn| {
-                    if let Some(channel_opening_params) = channel_opening_params {
-                        tracing::debug!(
-                            ?channel_opening_params,
-                            "Recording channel-opening parameters"
-                        );
-
-                        db::insert_channel_opening_params(conn, channel_opening_params)
-                            .map_err(SubmitOrderError::Storage)?;
-                    }
-
-                    runtime.block_on(orderbook_client.post_new_order(order.into()))
-                })
+    {
+        let mut db = db::connection().map_err(SubmitOrderError::Storage)?;
+        // We need to ensure that the channel-opening parameters are inserted after the
+        // order has been posted, but before order has been match and we need them to open
+        // the DLC channel. Without this database transaction we can easily end up with a
+        // match before we insert them!
```

Comment:
🤔 You have changed the code in a way that breaks what this comment is promisinh.

Maybe the original implementation was buggy, but by changing it I think we can now run into the problem outlined here.

In any case, you should delete the comment as part of the same patch (whether it is a good solution or not 😛).

Reply:
What would happen if we have channel opening params inserted before posting the order?

Reply:
I think you are right that it's okay to insert the data _before_ posting the order.

I'm not 100% sure if it could have any effect on attempting to open the channel a second time if the first one fails, so it might be worth checking. But I think it should be okay if this data is truly ephemeral.